reader/src/test/scala/za/co/absa/atum/reader/core/PartitioningIdProviderUnitTests.scala

```diff
@@ -38,7 +39,7 @@ class PartitioningIdProviderUnitTests extends AnyFunSuiteLike {
   private val atumPartitionsToNotFound = AtumPartitions(List.empty)
 
   private implicit val serverConfig: ServerConfig = ServerConfig(serverUrl)
-  private implicit val monad: IdMonad.type = IdMonad
+  private implicit val monad: MonadError[Identity] = IdMonad
```

Comment:
very clever-the whole file, just read it for the first time

Reply:
Also in line 71:

`val result: Long = response.getOrElse(throw new Exception("Failed to get partitioning id"))`

I think in tests this is unnecessary, I'd just do something like: `val result: Long = response.get`

Reply:
it's test that simulates something in well-isolated and very specific & deterministic scenario

Reply:
Line 91: isn't this a bit more readable?

`result.left.map(e => assert(e.isInstanceOf[ParsingException]))` instead of `swap`